packages/common/src/components/Settings/index.tsx

```diff
@@ -31,26 +34,30 @@ export const Settings = ({
         {connected && (
           <>
             <span>Wallet:</span>
-            {wallet?.publicKey && (
+            {publicKey && (
               <Button
                 style={{ marginBottom: 5 }}
-                onClick={() =>
-                  navigator.clipboard.writeText(
-                    wallet.publicKey?.toBase58() || '',
-                  )
-                }
+                onClick={async () => {
+                  if (publicKey) {
+                    await navigator.clipboard.writeText(publicKey.toBase58());
+                    notify({
+                      message: 'Wallet update',
+                      description: 'Address copied to clipboard',
+                    });
+                  }
+                }}
               >
                 <CopyOutlined />
-                {shortenAddress(wallet?.publicKey.toBase58())}
+                {shortenAddress(publicKey.toBase58())}
               </Button>
             )}
 
-            <Button onClick={select} style={{ marginBottom: 5 }}>
+            <Button onClick={open} style={{ marginBottom: 5 }}>
               Change
             </Button>
             <Button
               type="primary"
-              onClick={disconnect}
+              onClick={() => disconnect().catch()}
```

Comment:
Asynchronous background errors are handled by the context's global error handler.